tools/list.go

```diff
@@ -86,14 +95,30 @@ func (src List) Newest() (version.Number, List) {
 	return best, result
 }
 
+// Newest returns the greatest version in src, and the instances with that version.
+func (src Versions) Newest() (version.Number, Versions) {
+	var result Versions
+	var best version.Number
+	for _, agent := range src {
+		if best.Compare(agent.AgentVersion()) < 0 {
+			// Found new best number; reset result list.
+			best = agent.AgentVersion()
+			result = append(result[:0], agent)
+		} else if agent.AgentVersion() == best {
+			result = append(result, agent)
```

Comment:
do we wanna append or overwrite like above here?

Reply:
this is an exact copy of the tools code and I am reluctant to change such a low level bit of logic which could have unintended consequences